modules/nf-core/dada2/main.nf

```diff
@@ -0,0 +1,66 @@
+process DADA2 {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bioconductor-dada2:1.30.0--r43hf17093f_0':
+        'biocontainers/bioconductor-dada2:1.30.0--r43hf17093f_0' }"
+
+    input:
+    tuple val(meta), path(reads)
+
+    output:
+    tuple val(meta), path("*asvs.fasta")    , emit: dada2_out
+    path("*asv_counts.tsv")                 , emit: dada2_asv_counts
+    path "versions.yml"                     , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    if ( meta.single_end ){
+        """
```

Comment:
To avoid duplication below, follow this nf-core module example:
https://github.com/nf-core/modules/blob/master/modules/nf-core/bowtie2/align/main.nf
You can asign either "$reads" or "$reads[o] $reads[1]" to a variable depending on meta.single_end on groovy side, and then on the script side just write the module command and versions once.

Reply:
Done!